test: Apply pasta hack to scenarios as well

martinpitt opened the pull request:
Commit 9279fd46f1c1a73 didn't apply the hack to the "updates-testing" and "podman-next" scenarios, as these *overwrite* `VM_CUSTOMIZE_FLAGS` to get rid of the `--offline`.

Reorganize VM_CUSTOMIZE_FLAGS a bit to deal with `--no-network` in a cleaner way.

Fixes #1611

-----

I additionally triggered the scenarios here.

allisonkarlitskaya commented on Makefile:
```diff
@@ -158,11 +159,11 @@ rpm: $(TARFILE)
 
 ifeq ("$(TEST_SCENARIO)","updates-testing")
-VM_CUSTOMIZE_FLAGS = --run-command 'dnf -y update --setopt=install_weak_deps=False --enablerepo=updates-testing >&2'
+VM_CUSTOMIZE_FLAGS = $(PASTA_HACK) --run-command 'dnf -y update --setopt=install_weak_deps=False --enablerepo=updates-testing >&2'
```
Not enjoying the shadowing on this variable.  What's the story there?

martinpitt replied:
Mostly that we don't have a `--yes-network` to override the original `--no-network`

martinpitt replied:
we can probably do this with an `else` too, though

martinpitt replied:
Changed, WDYT?

allisonkarlitskaya reviewed (commented):
Hate to do this, but can you split this into two commits, one for the cleanup of the variable handling and one which we can revert later when podman gets fixed?

martinpitt commented:
@allisonkarlitskaya I'm afraid that's not quite possible any more, as the hack was already introduced in commit cecb2cc6e8f24 and modified in commit 9279fd46f1c1a737a . But I cleaned it up once more so that we only need to drop a single line (plus the comment) when it gets fixed.

allisonkarlitskaya reviewed (approved):
This looks a lot nicer, indeed.  Thanks!
